src/coreclr/vm/arm64/patchedcode.S

```diff
@@ -142,33 +142,80 @@ LOCAL_LABEL(ShadowUpdateEnd):
 #ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
     // Update the write watch table if necessary
     ldr  x12, LOCAL_LABEL(wbs_sw_ww_table)
-    cbz  x12, LOCAL_LABEL(CheckCardTable)
+    cbz  x12, LOCAL_LABEL(CheckCardTableBounds)
     add  x12, x12, x14, lsr #0xc  // SoftwareWriteWatch::AddressToTableByteIndexShift
     ldrb w17, [x12]
-    cbnz x17, LOCAL_LABEL(CheckCardTable)
+    cbnz x17, LOCAL_LABEL(CheckCardTableBounds)
     mov  w17, #0xFF
     strb w17, [x12]
 #endif
 
-LOCAL_LABEL(CheckCardTable):
-    // Branch to Exit if the reference is not in the Gen0 heap
+LOCAL_LABEL(CheckCardTableBounds):
+    // Branch to Exit if the reference is not in the heap
     ldr  x12, LOCAL_LABEL(wbs_ephemeral_low)
     ldr  x17, LOCAL_LABEL(wbs_ephemeral_high)
```

Comment:
btw, since you're here, here we have
```
    ldr  x12, LOCAL_LABEL(wbs_ephemeral_low)
    ldr  x17, LOCAL_LABEL(wbs_ephemeral_high)
```
can we change it to `ldp` ? is there an encoding that allows us to do it

Reply:
Sadly not.

The LDR is using the literal encoding, `LDR <Wt>, <label>`
https://docsmirror.github.io/A64/2023-09/ldr_lit_gen.html

LDP only has a `LDP <Wt1>, <Wt2>, [<Xn|SP>], #<imm>`, which isn't PC relative.
https://docsmirror.github.io/A64/2023-09/ldp_gen.html